src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/BKTree.cs

```diff
@@ -107,33 +107,39 @@ public void Find(ref TemporaryArray<string> result, string value, int? threshold
                 return;
             }
 
-            // We always want to compute the real edit distance (ignoring any thresholds).  This is
-            // because we need that edit distance to appropriately determine which edges to walk 
+            // We may need to compute the real edit distance (ignoring any thresholds) in the case
+            // where edges exist as we need that edit distance to appropriately determine which edges to walk 
             // in the tree.
             var characterSpan = currentNode.WordSpan;
+            var edgesExist = currentNode.EdgeCount > 0;
+
             var editDistance = EditDistance.GetEditDistance(
                 _concatenatedLowerCaseWords.AsSpan(characterSpan.Start, characterSpan.Length),
-                queryCharacters.AsSpan(0, queryLength));
+                queryCharacters.AsSpan(0, queryLength),
+                edgesExist ? int.MaxValue : threshold);
 
             if (editDistance <= threshold)
             {
                 // Found a match.
                 result.Add(new string(_concatenatedLowerCaseWords, characterSpan.Start, characterSpan.Length));
             }
 
-            var min = editDistance - threshold;
-            var max = editDistance + threshold;
-
-            var startInclusive = currentNode.FirstEdgeIndex;
-            var endExclusive = startInclusive + currentNode.EdgeCount;
-            for (var i = startInclusive; i < endExclusive; i++)
+            if (edgesExist)
```

Comment:
probably overkill to have this.  but i guess it doesn't hurt.

Reply:
I felt a little better about not having to think about a possible integer overflow otherwise.